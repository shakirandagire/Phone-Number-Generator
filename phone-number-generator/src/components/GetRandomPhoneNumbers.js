import React from 'react';

const GetRandomPhoneNumber = ({ minimum, maximum, total, phoneNumbers }) => (
  <div>
    {phoneNumbers.length > 0 && (
      <div>
        <div>
          <strong>Minimum Phone number: </strong>
          <span>{`0${minimum}`}</span>
        </div>
        <div>
          <strong>Maximum Phone number: </strong>
          <span>{`0${maximum}`}</span>
        </div>
        <div>
          <strong>Total Phone Numbers generated: </strong>
          <span>{total}</span>
        </div>
      </div>
    )}
  </div>
);
export default GetRandomPhoneNumber;
