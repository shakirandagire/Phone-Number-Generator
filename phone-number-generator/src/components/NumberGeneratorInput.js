import React from 'react';

const NumberGeneratorInput = ({ onClick, onChange }) => (
  <div>
  <h3 style={{margin: '20px'}}>Enter number of phone numbers to generate </h3>
    <form className="form-inline">
      <div className="form-group mx-sm-3 mb-2">
        <label htmlFor="input" className="sr-only">
          Enter Number
        </label>
        <input
          type="number"
          className="form-control"
          id="number"
          placeholder="Enter number"
          onChange={onChange}
        />
      </div>
      <button type="submit" className="btn btn-primary" onClick={onClick}>
        Generate Phone Numbers
      </button>
    </form>
  </div>
);
export default NumberGeneratorInput;