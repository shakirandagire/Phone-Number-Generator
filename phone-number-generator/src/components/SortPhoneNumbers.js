
import React from 'react';

const SortPhoneNumbers = ({ phoneNumbers, onChange }) => (
  <div className="sort">
    {phoneNumbers.length > 0 && (
      <div>
        <span className="text">
          <strong>Sort the phone numbers using:</strong>
        </span>
        {''}
        <select onChange={onChange} className="select">
          <option value="asc">Ascending</option>
          <option value="desc">Descending</option>
        </select>
      </div>
    )}
  </div>
);

export default SortPhoneNumbers;