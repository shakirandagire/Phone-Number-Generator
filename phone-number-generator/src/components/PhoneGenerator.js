import React, { Component } from 'react';
import NumberGeneratorInput from './NumberGeneratorInput';
import SortPhoneNumbers from './SortPhoneNumbers';
import ShowPhonesNumbers from './ShowPhonesNumbers';
import GetRandomPhoneNumber from './GetRandomPhoneNumbers'

class PhoneGenerator extends Component {
  state = {
    phoneNumbers: [],
    number: '',
    total: 0,
    sorter: 'asc',
    minimum: null,
    maximum: null,
    error: false
  };

  generatePhoneNumbers = event => {
    event.preventDefault();
    const { number, phoneNumbers } = this.state;
    if (!number || number > 10000)
      return this.setState({
        error: true,
        message: alert('The number hasnot been entered or exceeds the accepted number of phone numbers to be generated')
      });
      let phoneNumber = 0;
    while (phoneNumber < number) {
      phoneNumbers.push(
        '0' + Math.floor(Math.random() * 900000000 + 100000000)
      );
      phoneNumber++;
    }
    return this.setState(
      {
        phoneNumbers
      },
      () => this.getRandomPhoneNumbers(),
      this.handleSortingPhoneNumbers()
    );
  };

  getRandomPhoneNumbers= () => {
    const { phoneNumbers } = this.state;
    if (phoneNumbers.length > 0) {
      const minimum = Math.min(...phoneNumbers);
      const maximum = Math.max(...phoneNumbers);
      const total = phoneNumbers.length;
      this.setState({
        minimum,
        maximum,
        total
      });
    }
  };

  handleUserNumberInput = async event => {
    event.preventDefault();
    const number = event.target.value;
    if (Math.floor(Number(number))) {
      this.setState({
        number
      });
    }
  }

  handleSortingPhoneNumbers = () => {
    const { sorter, phoneNumbers } = this.state;
    if (!phoneNumbers.length > 0) return;
    if (sorter === 'asc') {
      this.setState({
        phoneNumbers: phoneNumbers.sort((x, y) => 0 - (x > y ? -1 : 1))
      });
    } else {
      this.setState({
        phoneNumbers: phoneNumbers.sort((x, y) => 0 - (x > y ? 1 : -1))
      });
    }
  };
  handleOnSortingChange = event => {
    event.preventDefault();
    const sorter = event.target.value;
    this.setState(
      {
        sorter
      },
      () => this.handleSortingPhoneNumbers()
    );
  };

  render() {
    const { minimum, maximum, total, phoneNumbers } = this.state;
    return (
        <div className="container">
            <div className="input">
              <NumberGeneratorInput
                onClick={this.generatePhoneNumbers}
                onChange={this.handleUserNumberInput}
              />
            </div>

            <GetRandomPhoneNumber
              phoneNumbers={phoneNumbers}
              minimum={minimum}
              maximum={maximum}
              total={total}
            />

            <SortPhoneNumbers
              phoneNumbers={phoneNumbers}
              onChange={this.handleOnSortingChange}
            />
            <ShowPhonesNumbers phoneNumbers={phoneNumbers} />
        </div>
    );
  }
}

export default PhoneGenerator;