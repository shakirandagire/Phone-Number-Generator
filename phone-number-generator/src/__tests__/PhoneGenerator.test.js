import React from 'react';
import enzyme from 'enzyme';
import NumberGeneratorInput from '../components/NumberGeneratorInput';
import ShowPhoneNumbers from '../components/ShowPhonesNumbers';
import GetRandomPhoneNumbers from '../components/GetRandomPhoneNumbers';
import SortPhoneNumbers from '../components/SortPhoneNumbers';
import PhoneGenerator from '../components/PhoneGenerator';

describe('PhoneGenerator Component', () => {
    it('should render properly', () => {
      const node = enzyme.shallow(<PhoneGenerator/>);
      expect(node).toMatchSnapshot();
    });
  });
describe('NumberGeneratorInput Component', () => {
  it('should render properly', () => {
    const node = enzyme.shallow(<NumberGeneratorInput />);
    expect(node).toMatchSnapshot();
  });
});

describe('ShowPhoneNumbers Component', () => {
    const props = {
        phoneNumbers: [],
       }
    it('should render properly', () => {
      const node = enzyme.shallow(<ShowPhoneNumbers {...props}/>);
      expect(node).toMatchSnapshot();
    });
  });

  describe('GetRandomPhoneNumbers Component', () => {
    const props = {
        phoneNumbers: [],
       }
    it('should render properly', () => {
      const node = enzyme.shallow(<GetRandomPhoneNumbers {...props}/>);
      expect(node).toMatchSnapshot();
    });
  });

  describe('SortPhoneNumbers Component', () => {
    const props = {
        phoneNumbers: [],
        onChange: jest.fn()
       }
    it('should render properly', () => {
      const node = enzyme.shallow(<SortPhoneNumbers {...props}/>);
      expect(node).toMatchSnapshot();
    });
  });