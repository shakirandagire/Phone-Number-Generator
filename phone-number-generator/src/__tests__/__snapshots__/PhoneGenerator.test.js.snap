// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetRandomPhoneComponent should render properly 1`] = `ShallowWrapper {}`;

exports[`GetRandomPhoneNumbers Component should render properly 1`] = `ShallowWrapper {}`;

exports[`Number Component should render properly 1`] = `ShallowWrapper {}`;

exports[`NumberGenerator Component should render properly 1`] = `ShallowWrapper {}`;

exports[`NumberGeneratorInput Component should render properly 1`] = `ShallowWrapper {}`;

exports[`NumberGeneratorInput Component should render properly 2`] = `ShallowWrapper {}`;

exports[`NumberGeneratorInput Component should render properly 3`] = `ShallowWrapper {}`;

exports[`NumberGeneratorInput Component should render properly 4`] = `ShallowWrapper {}`;

exports[`PhoneGenerator Component should render properly 1`] = `ShallowWrapper {}`;

exports[`ShowPhone Component should render properly 1`] = `ShallowWrapper {}`;

exports[`ShowPhoneNum ber Component should render properly 1`] = `ShallowWrapper {}`;

exports[`ShowPhoneNumbers Component should render properly 1`] = `ShallowWrapper {}`;

exports[`Sort Component should render properly 1`] = `ShallowWrapper {}`;

exports[`SortPhoneNumbers Component should render properly 1`] = `ShallowWrapper {}`;
